### PR TITLE
Fix #6248: Add New feature alert to notify existing users of the bottom toolbar

### LINKED ISSUE #6248: New features user education - bottom toolbar
### Description: 
Add UI (or new feature alert) to notify existing users of the bottom toolbar

UI TBC

@Sam-Sibley and @anthonypkeane to put the details in here

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

This pull request fixes #6248

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
Pre-Condition: Existing User + Bottom Bar Not Enabled

- Launch the App 
- Check Bottom Bar Alert
- Press Switch Bar
- Observe the Change

## Screenshots:

asatestt

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by soner-yuksel** [comment]
Blocked because of copy and localization but code and design is ready

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -1140,6 +1140,7 @@ public class BrowserViewController: UIViewController {
     // Priority: P3A - VPN - Default Browser - Rewards
     // TODO: Remove the dispatch after with a proper fix and fix calling present functions before super.viewDidAppear
     DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+      self.presentBottomBarCallout()
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController.swift]: I would say P3A should show first. What do you think?

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift [hunk 2]
```diff
@@ -31,6 +31,40 @@ extension BrowserViewController {
...
+    let onboardingNotCompleted =
+      Preferences.Onboarding.basicOnboardingCompleted.value != OnboardingState.completed.rawValue
+
+    if onboardingNotCompleted || !Preferences.General.isUsingBottomBar.value {
+      return
+    }
+
+    var bottomBarView = OnboardingBottomBarView()
+    bottomBarView.switchBottomBar = { [weak self] in
+      guard let self = self else { return }
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift]: you can use the fancy `guard let self else { return }` to make it shorter

File: Sources/Onboarding/Callouts/OnboardingBottomBarView.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import Shared
+import BraveUI
+import BraveShared
+
+public struct OnboardingBottomBarView: View {
```
Comment (iccub) [inline Sources/Onboarding/Callouts/OnboardingBottomBarView.swift:11]: Once we know more about v2, we can expand this view to be more generalized. Nothing for now

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift [hunk 4]
```diff
@@ -31,6 +31,40 @@ extension BrowserViewController {
...
+    if Preferences.DebugFlag.skipNTPCallouts == true || isOnboardingOrFullScreenCalloutPresented { return }
+
+    if presentedViewController != nil || !FullScreenCalloutManager.shouldShowDefaultBrowserCallout(calloutType: .bottomBar) {
+      return
+    }
+
+    let onboardingNotCompleted =
+      Preferences.Onboarding.basicOnboardingCompleted.value != OnboardingState.completed.rawValue
+
+    if onboardingNotCompleted || !Preferences.General.isUsingBottomBar.value {
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift]: `!Preferences.General.isUsingBottomBar.value` should not this be reversed? We want to show it only for existing users who do not have it enabled

**Review by iccub — CHANGES_REQUESTED** [review]

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift (hunk 4 above)
Comment (iccub) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift]: we should never show it to people who have bottom bar enabled, probably good to move to separate condition or use `AND` not `OR`

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (soner-yuksel) [reply Client/Frontend/Browser/BrowserViewController.swift]: Changed https://github.com/brave/brave-ios/pull/6640/commits/1b9470aca9072e5e1bcdfd337345da622ab950d1

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift (hunk 2 above)
Comment (soner-yuksel) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift]: Fixed at https://github.com/brave/brave-ios/pull/6640/commits/1b9470aca9072e5e1bcdfd337345da622ab950d1

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift (hunk 4 above)
Comment (soner-yuksel) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift]: Fixed and separated https://github.com/brave/brave-ios/pull/6640/commits/1b9470aca9072e5e1bcdfd337345da622ab950d1

**Review by iccub — APPROVED** [review]
nice :) lets wait for final copy and merge

**Comment by soner-yuksel** [comment]
The copy is updated.

**Pull request merged**